api/nameguard/nameguard.py

```diff
@@ -142,19 +143,25 @@ def analyse_label(self, label: str):
 
     async def inspect_name(
         self, network_name: NetworkName, name: str, resolve_labelhashes: bool = True, bulk_mode: bool = False
-    ) -> Union[NameGuardReport, ConsolidatedNameGuardReport]:
+    ) -> Optional[Union[NameGuardReport, ConsolidatedNameGuardReport]]:
         """
         Inspect a name. A name is a sequence of labels separated by dots.
         A label can be a labelhash or a string.
         If a labelhash is encountered and `resolve_labelhashes` is `True`, a lookup will be performed.
         """
+
+        logger.debug(f"[inspect_name] name: '{name}'")
+
+        # if len(name) > INSPECTABLE_NAMES_LENGTH: #TODO do we need it here or limit number of labelhashes in the name?
```

Comment:
@lightwalker-eth name length limit is checked after resolving labelhashes. Should we do it before or limit number of labelhashes in a name?

Reply:
@djstrong Good question. Perhaps we should define a different constant for the max number of labelhashes in a name that we might work to lookup? For example: Even 5 would be more than enough for 99.999999%+ of real cases.

If we're given a name with more labelhashes then that we could skip any lookups of those labelhashes.

What do you think?

Reply:
`MAX_NUMBER_OF_LABELHASHES_IN_NAME = 5`